### ISSUE TITLE
portamento and waveform

### ISSUE THREAD

**Issue by onukore** [kmatheussen/radium#670]
when i do portamento on sampler the sound change but not the  waveform, we don't have real size of waveform, if i do portamento up the sound finish beford waveform in editor

**Comment by kmatheussen** [comment]
Yes, the waveform is just an indication. There's lots of things that can make the waveform differ from the sound. I think only ADSR, various pitch parameters, and loop have an impact on the waveform.

**Comment by kmatheussen** [comment]
Sorry, I confused portamento with tremolo, for some reason. Portamento should definitely change waveform, yes.

**Comment by onukore** [comment]
this is bug ?

**Comment by kmatheussen** [comment]
I don't remember if this was implemented before 3.0, but with the current renderer, it has never been implemented.

**Comment by onukore** [comment]
in old version, it was i have sync sample in bar with portamento

**Comment by kmatheussen** [comment]
Which version is that?

**Comment by onukore** [comment]
in the first i have buy or beford

**Comment by kmatheussen** [comment]
I sort of vaguely remember that this worked some time, but when I look at the code, it is not implemented, and now I've also tried 1.9.42, 3.2.6, 3.9.6 and 4.0.8, and neither of them has waveform that matches portamento change, so I don't think it has ever been implemented.

**Comment by onukore** [comment]
maybe i confuse with other sorry

**Comment by onukore** [comment]
we don't need waveform move, juste recalculate length, for the end of sound be the end of waveform

**Comment by kmatheussen** [comment]
> we don't need waveform move, juste recalculate length, for the end of sound be the end of waveform

Unfortunately, that's not trivial. Same with sample seek.

**Comment by kmatheussen** [comment]
To clarify: It's not trivial to calculate sample positions and length sample-accurately, but at least for displaying waveform, it doesn't have to be that accurate. And I guess this is mostly true for sample seek as well; It's probably better to do a sample seek where the samples might be played a little bit from the wrong offset positions, than to not have sample seek at all.

**Comment by onukore** [comment]
yes i think too
